### PR TITLE
doc: publish initial user docs as github pages with jekyll gitbook theme

### PR DESCRIPTION
**Pull request by alexbrdn**
Check https://wot-oss.github.io/tmc/ for the current version

After merging the PR, change setup of github pages deployment to deploy from the main branch

### CODE REVIEW

**Comment by hadjian** [comment]
Very nice start. Thx. Aside from the few comments LGTM.

File: docs/_pages/40-workflows.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+You can also use any directory as a storage space for an unnamed local repository. You will need to pass a `--directory` flag to most commands.
+
+## Expose a Catalog for HTTP Clients
+
+To expose a catalog over HTTP, start a server:
+```bash
+tmc serve
+```
+The OpenAPI description of the API [is available][3] in our GitHub repository.
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply docs/_pages/40-workflows.md:71]: done

File: docs/_pages/50-commands.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+HTTP repo type allows accessing a file repo exposed over HTTP by simple directory hosting. It also permits
+accessing a repo committed to some git forge, e.g. GitHub or GitLab.
+
+For repos of type "http", 'loc' field is the URL of the repository.
+
+HTTP repos are read-only.
+
+The structure of HTTP directory tree is expected to be exactly as the one stored by a file repo.
+That means, that normally, the relative file paths are simply appended to the 'loc' URL. If the server requires putting
+the requested file's path anywhere other than at the end of the URL, you can use a placeholder `{{ID}}` in the URL, both in the path and in query part.
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply docs/_pages/50-commands.md:44]: fixed

File: docs/_pages/50-commands.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+E.g. for an HTTP repo served by a GitLab installation, you may use ```https://example.com/api/v4/projects/<project-id>/repository/files/{{ID}}?ref=main```
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply docs/_pages/50-commands.md:46]: fixed

File: README.md [hunk 4]
```diff
@@ -7,54 +7,45 @@
...
-tmc completion <shell> -h
-```
+    ```bash
+    tmc completion <shell> -h
+    ```
 
-### Browse the canoncial Catalog
+### Browse the Canonical Catalog
 
 1. Configure the canonical repository
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply README.md]: I've rephrased this.

File: docs/_pages/50-commands.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+Basic usage of `attachment fetch` is straightforward, however the `--concat` flag requires some elaboration.
+
+### `--concat`
+
+The `--concat` flag allows automating the production of documentation of changes between versions of TM. For example, you may put
+your general description of a device into a file "README.md" and attach it to the TM name. Then, for each version of 
+TM you import into the catalog, you may put just the version-specific documentation or changelog in an attachment with the same
+name "README.md" and attach it to the corresponding TM id. Given this setup, if you then use execute `tmc attachment fetch --concat <tm-name> README.md`
+you will get a file, which is a concatenation of the "README.md" attached to TM name and all "README.md" files attached to
+individual TMs with this TM name.
```
[the remark this answers is not in the document]
Comment (alexbrdn) [reply docs/_pages/50-commands.md:69]: added TM name and TM version to glossary

**Comment by alexbrdn** [comment]
@hadjian, please review the updates

**Comment by hadjian** [comment]
> @hadjian, please review the updates

The changes LGTM. I feel 'Create a catalog' is missing?

**Comment by alexbrdn** [comment]
@hadjian, I've added tips on how to host a catalog. Pls review

**Comment by hadjian** [comment]
> @hadjian, I've added tips on how to host a catalog. Pls review

Found one more line about the canonical repo: https://github.com/wot-oss/tmc/blob/c71bade0257714bebd2d9c625b4a4415fdf94025/docs/_pages/20-gettingstarted.md?plain=1#L9

About the workflow "Publish a Catalog to a Git Forge": to be able to do this, how is the user supposed to know how to create a file repository?

**Comment by alexbrdn** [comment]
I've removed the offending sentence altogether.

> About the workflow "Publish a Catalog to a Git Forge": to be able to do this, how is the user supposed to know how to create a file repository?
https://github.com/wot-oss/tmc/blob/doc/add-user-doc/docs/_pages/50-commands.md#repo-add

**Pull request merged**